include/ofi_proto.h

```diff
@@ -92,20 +85,28 @@ enum {
  *     Tx side includes in subsequent packets.  This field is used for
  *     rendezvous protocol.
  *     The rx_key may be formed similar to message_id.
- * seg_num: This is the total number of segments that is sent by sender
- *     in a segmentation and reassembly (SAR) protocol.
+ * ctrl_data: This is provider specific flags for remote side
+ * seg_size:
+ *     Data packets - size of current message, in bytes.
+ *     Large data packets - size of current message, 2 ^ seg_size, in bytes
+ *     Ctrl packets - number of segments in window allowed past seg_no.
+ * seg_no:
+ *     Data packets - position 0..(n-1) of segment in current message.
+ *     Ctrl packets - last segment ack'ed.
  */
 struct ofi_ctrl_hdr {
-	uint8_t			version;
-	uint8_t			type;
-	uint16_t		seg_size;
-	uint32_t		seg_no;
-	uint64_t		conn_id;
-	uint64_t		msg_id;
+	uint8_t				version;
+	uint8_t				type;
+	uint64_t			conn_id;
```

Comment:
This is introducing undefined padding into the wire structure.  We're not saving anything by moving the seg_size and seg_no into the union.

Reply:
Yes, you're right. Thanks. Fixed